gapic/src/main/com/google/gapid/models/Settings.java

```diff
@@ -224,10 +233,27 @@ private static int getInt(Properties properties, String name, int dflt) {
     }
   }
 
+  private static long getLong(Properties properties, String name, long dflt) {
+    String value = properties.getProperty(name);
+    if (value == null) {
+      return dflt;
+    }
+
+    try {
+      return Long.parseLong(value);
+    } catch (NumberFormatException e) {
+      return dflt;
+    }
+  }
+
   private static boolean getBoolean(Properties properties, String name) {
     return "true".equalsIgnoreCase(properties.getProperty(name, ""));
   }
 
+  private static boolean getBoolean(Properties properties, String name, boolean dflt) {
```

Comment:
Seems like a bit of a long-winded solution for the sake of a `true` default. I'd buy this argument more if other types didn't have non-default-initialized values. `noAutoUpdates` reads like it would apply the update automatically, so it would probably end up named as `noAutoCheckForUpdates`. Then you'd end up with a number of conditionals being a double-negative: `if (!noAutoCheckForUpdates)`, which I've seen cause countless bugs in my career.